.ci/setup.sh

```diff
@@ -46,16 +46,19 @@ setup_distro_env() {
 	sudo systemctl start haveged
 }
 
-install_docker() {
+install_container_engine() {
 	if [ "${TEST_CGROUPSV2}" == "true" ]; then
 		info "Docker won't be installed: testing cgroups V2"
 		return
 	fi
 
-	if ([ "$arch" == "ppc64le" ] || [ "$arch" == "s390x" ]) && ([ "$ID" == "fedora" ] || [[ "${ID_LIKE:-}" =~ "fedora" ]]); then
-		# download.docker.com does not package for ppc64le and s390x and on
-		# Fedora-likes, it's not packaged by the distro either. Use Podman.
-		sudo dnf install -y podman runc
+	if [ "$ID" == "fedora" ] || [[ "${ID_LIKE:-}" =~ "fedora" ]]; then
+		# Podman is the primary container engine on Fedora-likes
+		# Try reinstalling to fix CNI configuration
+		sudo dnf reinstall -y podman || sudo dnf install -y podman
```

Comment:
Actually reinstalling also fixes the inability to build Ubuntu (errors are reminiscent of https://github.com/containers/podman/issues/9421), so there's something deeper that should be fixed here.